addons/point_of_sale/models/pos_config.py

```diff
@@ -8,33 +8,67 @@
 from odoo.exceptions import ValidationError
 
 
-class AccountCashboxLine(models.Model):
-    _inherit = 'account.cashbox.line'
-
-    default_pos_id = fields.Many2one('pos.config', string='This cashbox line is used by default when opening or closing a balance for this point of sale')
-
-    @api.multi
-    def name_get(self):
-        result = []
-        for cashbox_line in self:
-            result.append((cashbox_line.id, "%s * %s"%(cashbox_line.coin_value, cashbox_line.number)))
-        return result
-
 class AccountBankStmtCashWizard(models.Model):
     _inherit = 'account.bank.statement.cashbox'
 
+    @api.depends('pos_config_ids')
+    def _compute_currency(self):
+        super(AccountBankStmtCashWizard, self)._compute_currency()
+        for cashbox in self:
+            if cashbox.pos_config_ids:
+                cashbox.currency_id = cashbox.pos_config_ids[0].currency_id.id
+
+    currency_id = fields.Many2one('res.currency', compute='_compute_currency')
+    pos_config_ids = fields.One2many('pos.config', 'default_cashbox_id')
+    is_a_template = fields.Boolean(default=False)
+
     @api.model
     def default_get(self, fields):
         vals = super(AccountBankStmtCashWizard, self).default_get(fields)
+        if "is_a_template" in fields and self.env.context.get('default_is_a_template'):
+            vals['is_a_template'] = True
         config_id = self.env.context.get('default_pos_id')
         if config_id:
-            lines = self.env['account.cashbox.line'].search([('default_pos_id', '=', config_id)])
+            config = self.env['pos.config'].browse(config_id)
+            previous_session_list = self.env['pos.session'].search([('config_id', '=', config_id)])
+            if len(previous_session_list) > 1 and previous_session_list[1].cash_register_id.cashbox_end_id.cashbox_lines_ids:
```

Comment:
shouldn't you use 'last_session_closing_cash' on config, set in compute: '_compute_last_session'.

If it doesn't work, I think we should use it and adapt it.